lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -65,13 +89,64 @@ public void OnNext(IContextMessage contextMessage)
                     //// For the counters from multiple evaluators with the same counter name, the value should be aggregated here
                     //// We also need to consider failure cases.  
                     _counters[counter.Name] = counter;
+
+                    if (_totalIncrementSinceLastSink.ContainsKey(counter.Name))
+                    {
+                        _totalIncrementSinceLastSink[counter.Name] += counter.Value - c.Value;
+                    }
+                    else
+                    {
+                        _totalIncrementSinceLastSink.Add(counter.Name, counter.Value - c.Value);
+                    }
```

Comment:
A counter keeps its current value only. It send the current counter data with heartbeat. It is up to the consumer for when and how to process it.  If counter itself keeps last increment, that means the counter needs to know how it is consumed. What if there are multiple consumers? Also in this case, the consumer is at driver side, it won't sent the updated  counter back to evaluator. So I would still like to keep the counter simple and let the consumer to keep the data for how and when it is processed. .